From fe39e44858fa443047b49d013ae82ccd5ee9d4ef Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Tue, 4 Aug 2020 11:39:46 -0500
Subject: [PATCH 23/23] imx8mm_beacon: Update DTB for I-Temp vs C-Temp parts

Industrial temp parts can operate at higher temperatures than regular
commercial temperature parts, but the device tree is configured for
commercial temperatures.

This patch checks to see if the part is Industrial temp, and if
so, it will automatically update the DTB to reflect proper trip points

Fixes: PHOENIX8-176

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 board/beacon/imx8mm/imx8mm_beacon.c | 7 +++++++
 include/configs/imx8mm_beacon.h     | 6 +++++-
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/board/beacon/imx8mm/imx8mm_beacon.c b/board/beacon/imx8mm/imx8mm_beacon.c
index da84aecdf5..1f0ee8320e 100644
--- a/board/beacon/imx8mm/imx8mm_beacon.c
+++ b/board/beacon/imx8mm/imx8mm_beacon.c
@@ -27,6 +27,7 @@
 #include <imx_mipi_dsi_bridge.h>
 #include <mipi_dsi_panel.h>
 #include <asm/mach-imx/video.h>
+#include <imx_thermal.h>
 
 DECLARE_GLOBAL_DATA_PTR;
 
@@ -521,10 +522,16 @@ int board_mmc_get_env_dev(int devno)
 
 int board_late_init(void)
 {
+	int minc, maxc;
 #ifdef CONFIG_ENV_IS_IN_MMC
 	board_late_mmc_env_init();
 #endif
 
+	if (get_cpu_temp_grade(&minc, &maxc) == TEMP_INDUSTRIAL) {
+		env_set("trip0", "0x17318");
+		env_set("trip1", "0x19a28");
+	}
+
 	return 0;
 }
 
diff --git a/include/configs/imx8mm_beacon.h b/include/configs/imx8mm_beacon.h
index 165c09ebe7..e8e1ab7a00 100644
--- a/include/configs/imx8mm_beacon.h
+++ b/include/configs/imx8mm_beacon.h
@@ -139,8 +139,12 @@
 	"loadbootscript=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
+	"updatefdt=fdt addr $fdt_addr; " \
+	"  fdt set /thermal-zones/cpu-thermal/trips/trip0 temperature < ${trip0} >; " \
+	"  fdt set /thermal-zones/cpu-thermal/trips/trip1 temperature < ${trip1} > \0"\
 	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
-	"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
+	"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}; " \
+	"  if test -n ${trip1} > 0 ; then run updatefdt; fi\0" \
 	"loadramdisk=load mmc ${mmcdev} ${ramdisk_addr} ${ramdiskimage}\0"\
 	"mmcboot=echo Booting from mmc ...; " \
 		"run finduuid; run mmcargs; " \
-- 
2.17.1

